### PR TITLE
fix(CI): Daily CI with Dafny 4.1.0

### PR DESCRIPTION
**Pull request by texastony**
*Issue #, if available:* Test the MPL daily with Dafny 4.1.0

*Description of changes:*
Break the GitHub workflows up into two distinct groups:
1. Workflows that define jobs that test our product
(`library_{dafny_verification, LANG_tests}.yml`, `duvet.yml`, etc.) 
2. Workflows that utilize those in (1.) ON a condition:
(`daily_ci.yml`, `pull.yml`, `push.yml`, `nightly_dafny.yml`)

This gives us better control of what actions run when.
Best of all, we can have a build badge on Push or Daily CI. 

*Squash/merge commit message, if applicable:*
`fix(CI): Daily CI with Dafny 4.1.0`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/daily_ci.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# This workflow runs every weekday at 15:00 UTC (8AM PDT)
+name: Daily CI
```
Comment (robin-aws) [inline .github/workflows/daily_ci.yml:2]: Just curious, is the purpose of this to catch spontaneous build breaks even when there are no active PRs? Ideally with reproducible builds this wouldn't be necessary, but I appreciate that reality can be less than ideal. :)

**Review by robin-aws — DISMISSED** [review]
Much cleaner, thanks! I was considering doing a similar refactoring myself, and I think we should encourage this pattern for Dafny projects in general.

**Comment by robin-aws** [comment]
Resolved the conflicts by ignoring my own edits - I do think it would be useful to trigger the nightly build for a given Dafny version, but that will require a bit more GHA coding so I'd rather not block on it now, and just add it if and when we want it later.

**Review by robin-aws — APPROVED** [review]

File: .github/workflows/daily_ci.yml (hunk 1 above)
Comment (texastony) [reply .github/workflows/daily_ci.yml:2]: I have a long standing ambition that every Crypto Tools Product executes its CI at least once every week day.
A failure of this execution would cut a Sev3 ticket.

I want this.
One Day, I will get it. 

This is a start.
It also let's us have a clean build badge.

**Comment by texastony** [comment]
> Resolved the conflicts by ignoring my own edits - I do think it would be useful to trigger the nightly build for a given Dafny version, but that will require a bit more GHA coding so I'd rather not block on it now, and just add it if and when we want it later.

Thanks @robin-aws!
All of the Testing Jobs take the Dafny Version as an input. 
We are "unblocked" from triggering this jobs on a Given Dafny Version. 
Furthermore, it should be trivial to author a PR to add another GH Workflow that tests a particular Dafny. 

Just expand the `.github/workflows/pull.yml` Workflow with 3 more Jobs,
setting the Dafny version appropriately.

**Comment by robin-aws** [comment]
> Just expand the .github/workflows/pull.yml Workflow with 3 more Jobs,
setting the Dafny version appropriately.

Actually, would it make more sense to make a separate `.github/workflows/manual.yml` workflow with only the manual trigger?

**Pull request merged**